src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.DebugVerifier.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class NullableWalker
+    {
+#if DEBUG
+        private sealed class DebugVerifier : BoundTreeWalker
+        {
+            private static ImmutableArray<BoundKind> s_skippedExpression = ImmutableArray.Create(BoundKind.ArrayInitialization, BoundKind.ObjectInitializerExpression, BoundKind.CollectionInitializerExpression, BoundKind.DynamicCollectionElementInitializer);
+            private readonly PooledDictionary<BoundExpression, (NullabilityInfo Info, TypeSymbol Type)> _analyzedNullabilityMap;
+            private readonly HashSet<BoundExpression> _visitedNodes = new HashSet<BoundExpression>();
+            private int _recursionDepth;
+
+            private DebugVerifier(PooledDictionary<BoundExpression, (NullabilityInfo Info, TypeSymbol Type)> analyzedNullabilityMap)
+            {
+                _analyzedNullabilityMap = PooledDictionary<BoundExpression, (NullabilityInfo Info, TypeSymbol Type)>.GetInstance();
```

Comment:
>_analyzedNullabilityMap [](start = 16, length = 23)

I didn't understand why we made a copy of the map. Can't we use the original one? The `DebugVerifier` doesn't seem to modify the map.
If you want a readonly copy, consider `ToImmutableDictionary`.